### PR TITLE
chore: bump notebook-controller to 1.9.0-rc.0

### LINKED ISSUE #88: Update `notebook-controller` ROCK
### Context

Each charm in Kubeflow has a ROCK (or a set of ROCKs) it is using. The process of upgrading ROCKs usually means going to the component's upstream repository Dockerfile, comparing current ROCK version against the one in the repository and adding the missing bits.

### What needs to get done

1. Clone the [upstream's](https://github.com/kubeflow/kubeflow) kubeflow repository.
2. Go to desired Dockerfile (first line in `rockcraft.yaml` should specify the full path).
3. Checkout to the branch or tag of the target version.
4. Checkout to the branch or tag of the version of the current ROCK.
5. Compare both files to spot the differences.

### Definition of Done

1. New ROCK is ready.
2. Tests are passing.
3. ROCK is published under `charmedkubeflow` Dockerhub account.

### PR DESCRIPTION
**Pull request by orfeas-k**
IIRC we generally do not include the `rc` part of the tag in the image's version.

Closes #88

### CODE REVIEW

File: notebook-controller/tests/__pycache__/test_rock.cpython-38-pytest-8.2.2.pyc
Comment (misohu) [inline notebook-controller/tests/__pycache__/test_rock.cpython-38-pytest-8.2.2.pyc:1]: I dont think we need this file.

File: notebook-controller/rockcraft.yaml
Comment (misohu) [inline notebook-controller/rockcraft.yaml:1]: There were not so many changes in this file ... but I believe we should rewrite this rockcraft as follows:
- we need to specify exact go version ... this rock will use latest for this part 
- we should separate security requirements in separate part as we do in all other rocks 

For both you can take inspiration e.g. in this rockcraft https://github.com/canonical/kserve-rocks/blob/KF-5718-update-agent-v-0-12-1/agent/rockcraft.yaml

**Review by misohu — COMMENTED** [review]
Little comments

File: notebook-controller/tests/__pycache__/test_rock.cpython-38-pytest-8.2.2.pyc
Comment (orfeas-k) [reply notebook-controller/tests/__pycache__/test_rock.cpython-38-pytest-8.2.2.pyc:1]: Removed

File: notebook-controller/rockcraft.yaml
Comment (orfeas-k) [reply notebook-controller/rockcraft.yaml:1]: Good catches, updated!

**Review by misohu — APPROVED** [review]
LGTM

**Pull request merged**